cloudevents/spec.md

```diff
@@ -178,6 +178,12 @@ CloudEvents attribute names MUST consist of lower-case letters ('a' to 'z') or
 digits ('0' to '9') from the ASCII character set. Attribute names SHOULD be
 descriptive and terse and SHOULD NOT exceed 20 characters in length.
 
+#### Reserved Names
+
+The following attribute names are reserved and MAY NOT be used:
```

Comment:
without additional text I can imagine someone being confused because in https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#event-data it talks about `data` as if it were a context attribute, even though (I guess) technically it is not. But I suspect most casual readers of the spec won't pick up on the distinction.

Perhaps:
- - -
- `data` : while technically not a "context attribute" this name is reserved for cases where the event payload needs to appear along-side the context attributes. For example, in JSON structured mode.
- - -

Also, feels odd to have an entire section for just one and I wonder if we'll ever have more?? If we have any more I suspect we'll add them as optional attributes won't we? Maybe just add a sentence to the "Event Data" section explaining that "data" is special?

Reply:
@duglin: I think all we need to do is remove "When present, this information will be encapsulated within `data`." in the event-data section - and it would be good to do that in this PR to make it a consistent change.

I do think we need to prevent `data` from being used *within the context attributes section* but I agree it probably doesn't need a new subsection.

I would propose:

- Renaming "Attribute Naming Convention" to "Attribute Naming" (the section contains normative requirements)
- At the end of the current second paragraph in that section, append: "Context attributes MUST NOT have the name `data`; this name is reserved as it is used for CloudEvent data in some event formats in places which would otherwise be ambiguous."

Reply:
If it's easiest for me to come up with a counter-PR, let me know :)

Reply:
Let me take a crack at this..